### DISCUSSION TITLE
Integration with nlohmann_json using Cmake

### DISCUSSION THREAD

**Discussion by AntonioRasulo** [pboettch/json-schema-validator#247] [Q&A]
Hi, I'm having issues by integrating the schema-validator library with the json nlohmann library. 
Here's the snippet of interest of the CMakeLists.txt that I wrote:

```Cmake
########################################################################
# PROJECT LIBS
########################################################################

add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/lib/json)

add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/lib/json-schema-validator)

########################################################################
# PROJECT EXE
########################################################################

add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/src)
```

where the first `add_subdirectory`  is for the nlohmann json library, the second one is for the schema validator library and the third one is for my project. It appear the following error:

    CMake Error: install(EXPORT "nlohmann_json_schema_validatorTargets" ...) includes target "nlohmann_json_schema_validator" 

which requires target "nlohmann_json" that is not in any export set.				

The install function of interest Inside the CMakeLists.txt of the schema validator library seems to be this one:

```CMake
if(JSON_VALIDATOR_INSTALL)
    # Set the install path to the cmake config files (Relative, so install works correctly under Hunter as well)
    set(INSTALL_CMAKE_DIR "lib/cmake/${PROJECT_NAME}")
    set(INSTALL_CMAKEDIR_ROOT share/cmake)

    # Install Targets
    install(EXPORT ${PROJECT_NAME}Targets
            FILE ${PROJECT_NAME}Targets.cmake
            DESTINATION "${INSTALL_CMAKE_DIR}")`
```

What I have to do to solve this problem?

**Comment by pboettch** [comment]
Do you need to install the libraries (and the sub-directories)? If not, just defined

```CMake
set(JSON_VALIDATOR_INSTALL OFF)
```

before including json-schema-validator.

**Reply by AntonioRasulo** [reply]
Yes I need to install them. I want to use your library and the nlohmann json library in my project.
